### ISSUE TITLE
Bad performance after switching to eslint flatConfig

### ISSUE THREAD

**Issue by RARgames** [webpack-contrib/eslint-webpack-plugin#267]
### Bug report

Bad performance after switching to eslint flatConfig

### Actual Behavior

Before changing my load time of the app was around 18.5s, and subsequent was ~6.8s.
I updated my code to use flatConfigs (config is slightly different, but lint time differences are < .2s).
Now my initial loading time is about 25s and subsequent time almost doubled to 12.7s.

### Expected Behavior

The performance should be similar using similar configs. Doubling the time means that something is clearly wrong.

### How Do We Reproduce?

I think the best way is just to take some recommended eslint configs (before and flatConfigs later). Can provide my exact case later if helpful.

### Please paste the results of `npx webpack-cli info` here, and mention other relevant information
```
System:
    OS: Windows 10 10.0.19045
    CPU: (32) x64 Intel(R) Core(TM) i9-14900K
    Memory: 41.52 GB / 63.83 GB
  Binaries:
    Node: 20.11.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.8.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (127.0.2651.74)
```

**Comment by RARgames** [comment]
It's possible to use `lintDirtyModulesOnly: true,`, which drastically improves times, but it works differently. If I want to check all files it was much faster using old configs.

**Comment by alexander-akait** [comment]
We don't use something special for `flatConfig`, it means performance regression on `eslint` side

**Comment by alexander-akait** [comment]
Check our code

**Comment by alexander-akait** [comment]
Most likely you made a mistake in the migration, I advise you to try to run without webpack and  check performance too

**Comment by RARgames** [comment]
I know that. I did more testing and...
I can get similar times as before without `lintDirtyModulesOnly: true`

I created a new flat config in just one file with 1 set of rules combining all the flatConfigs and it works perfectly. It seems that the main issue is combining flatConfigs by eslint.

The eslint performance (without starting the app) seems to be stable, maybe even slightly better - checked it using `TIMING=all` calculated sums and compared. There is not much difference old vs flat config.

Not sure how to resolve this fully now, but I achieved some improvements that is enough at least for now (I'll think of better solution).
Thanks for help
